Applications/AdminApi/test/AdminApi.Tests.Integration/StepDefinitions/ClientDetailsStepDefinitions.cs

```diff
@@ -69,6 +69,7 @@ public async Task ThenTheResponseContainsAClient()
         _response!.Result!.ClientId.Should().Be(_clientId);
         _response!.Result!.DefaultTier.Should().Be(_tierId);
         _response!.Result!.MaxIdentities.Should().Be(_maxIdentities);
+        _response!.Result!.NumberOfIdentities.Should().BeLessOrEqualTo(_maxIdentities);
```

Comment:
This is repetition of business logic. Also, it's logic that isn't even related to this test.

I will remove it.

Reply:
...that is why I added it here, so that it makes life easier for anyone in the future who may be in the dilemma whether it is necessary or not.

Reply:
I see. 

I would expect our schema validator to uncover such problems (=> additional properties that don't exist on the type). Since that seems to not be the case, we should rather fix it there.

Alternatively, as a short-term solution, we could have checked with `.Should().NotBeNull()`. This lacks any business logic and still makes sure that what you describe cannot happen.
But let's not do that now, as this would trigger another pipeline run. And I finally want to merge this. :)
I will add the improvement of the schema validator to our backlog.